kernel/include/kswap.h

```diff
@@ -96,10 +96,10 @@ static ALWAYS_INLINE unsigned int do_swap(unsigned int key,
 	 * have it.  We "release" other spinlocks here.  But we never
 	 * drop the interrupt lock.
 	 */
-	if (is_spinlock && lock != NULL && lock != &sched_spinlock) {
+	if (is_spinlock != 0 && lock != NULL && lock != &sched_spinlock) {
```

Comment:
Should `is_spinlock` be changed to `bool` instead?

Reply:
This was considered, and it would actually be a better solution. However, is_spinlock is a function parameter, and I do not feel authorized to change interfaces.

Reply:
Then I suggest renaming parameter and creating a local `bool is_spinlock` and assigning that at function entry instead of sprinkling this throughout the code.

Reply:
> This was considered, and it would actually be a better solution. However, is_spinlock is a function parameter, and I do not feel authorized to change interfaces.

Zephyr is an open project, and you don't need any authorisation to suggest code changes through PRs (it's another matter whether they'll get accepted). I think an API change would be justified here. If it's determined that an API change is not desired, then a local `bool` variable like @olofj suggests is the second best option (IMO).

Reply:
is_spinlock parameter has been changed to bool now.